openerp/models.py

```diff
@@ -385,9 +385,40 @@ def _field_create(self, cr, context=None):
                     (name_id, context['module'], 'ir.model', model_id)
                 )
 
-        cr.execute("SELECT * FROM ir_model_fields WHERE model=%s", (self._name,))
+        # OpenUpgrade edit start: In rare cases, an old module defined a field
+        # on a model that is not defined in another module earlier in the
+        # chain of inheritance. Then we need to assign the ir.model.fields'
+        # xmlid to this other module, otherwise the column would be dropped
+        # when uninstalling the first module.
+        # An example is res.partner#display_name defined in 7.0 by
+        # account_report_company, but now the field belongs to the base
+        # module
+        # Given that we arrive here in order of inheritance, we simply check
+        # if the field's xmlid belongs to a module already loaded, and if not,
+        # drop the record to let the following standard code create it for the
```

Comment:
@pedrobaeza thanks! sorry for not getting it in the first place, the day was long...